Connect4.hs

```diff
@@ -159,4 +159,49 @@ randomPlayer (State _ avail) = pickRandomAction [Action e | e <- [1..7], Action
 pickRandomAction :: [Action] -> IO Action
 pickRandomAction availActions = do
     pickIndex <- randomRIO (0, length availActions - 1)
-    return (availActions !! pickIndex)
\ No newline at end of file
+    return (availActions !! pickIndex)
+
+-- monteCarloPlayer plays out each move X times by selecting random moves until completion, and selects the best outcome. (Currently only checks if game end in win or loss)
+monteCarloPlayer ::  Integer -> Player
+monteCarloPlayer numGames state = do
+    actionValuePair <- mc connect4 state numGames
+    return (fst actionValuePair)
+
+-- For each action, play out X games and return the pair with the highest expected value (most wins)
+mc:: Game -> State -> Integer -> IO (Action, Double)
+mc game st numGames = do
+    actionValuePairs <- getIOValues [ mcPlayOutXGames game st action numGames | action <- avail]
+    putStrLn (show actionValuePairs)
```

Comment:
Do we want this? If so, we should add more text about what it means

Reply:
nah we can remove it, just for testting